src/lib/OS/SimOS.cc

```diff
@@ -133,22 +143,7 @@ void SimOS::tick() {
   }
 }
 
-const std::shared_ptr<Process>& SimOS::getProcess(uint64_t TID) {
-  auto proc = processes_.find(TID);
-  if (proc == processes_.end()) {
-    // If TID doesn't exist then hard exit
-    std::cerr << "[SimEng:SimOS] ERROR : Process with TID `" << TID
-              << "` does not exist.\n";
-    exit(1);
-  }
-  return proc->second;
-}
-
-void SimOS::createInitialProcess() {
-  // TODO : When supporting multiple processes, need to keep track of next
-  // available TGID and TIDs, and pass these when constructing a Process
-  // object
-
+uint64_t SimOS::createProcess(std::optional<span<char>> instructionBytes) {
```

Comment:
This can be simplified by using a span created by the default constructor as the default value in the function i.e 
`uint64_t SimOS:createProcess(span<char> instrBytes = span<char>())`
then 
`if (instructionBytes.has_value()) {}` can be replaced with `if(!instructionBytes.empty()) {}`

This will also replace calls make to `createProcess({})` with `createProcess()`. IMO it is much cleaner and doesn't require `std::optional`

Reply:
Done